src/Test/Unit/WarmUp/UrlsPattern/CommandArgumentBuilderTest.php

```diff
@@ -0,0 +1,125 @@
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritDoc
+ */
+class CommandArgumentBuilderTest extends TestCase
+{
+    /**
+     * @var CommandArgumentBuilder
+     */
+    private $argumentBuilder;
+
+    /**
+     * @var LoggerInterface|MockObject
+     */
+    private $loggerMock;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUp()
+    {
+        $this->loggerMock = $this->getMockForAbstractClass(LoggerInterface::class);
+
+        $this->argumentBuilder = new CommandArgumentBuilder($this->loggerMock);
+    }
+
+    /**
+     * @param string $entity
+     * @param string $storeIds
+     * @param array $expected
+     * @dataProvider generateDataProvider
+     */
+    public function testGenerate(string $entity, string $storeIds, array $expected)
+    {
+        $this->loggerMock->expects($this->never())
+            ->method('info');
+
+        $this->assertEquals(
+            $expected,
+            $this->argumentBuilder->generate($entity, $storeIds)
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function generateDataProvider(): array
+    {
+        return [
+            [
+                'category',
+                'store_code',
+                [
+                    '--entity-type=category',
+                    '--store-id=store_code',
+                ],
+            ],
+            [
+                'category',
+                '1|2|3',
+                [
+                    '--entity-type=category',
+                    '--store-id=1',
+                    '--store-id=2',
+                    '--store-id=3',
+                ],
+            ],
+            [
+                'cms-page',
+                'Store Code 1|Store Code 2|Store Code 3',
+                [
+                    '--entity-type=cms-page',
+                    '--store-id=Store Code 1',
+                    '--store-id=Store Code 2',
+                    '--store-id=Store Code 3',
+                ],
+            ],
+        ];
+    }
+
+    public function testGenerateWithProductSkus()
+    {
+        $this->loggerMock->expects($this->never())
+            ->method('info');
+
+        $this->assertEquals(
+            [
+                '--entity-type=product',
+                '--store-id=store_1',
+                '--store-id=store_2',
+                '--product-sku=sku1',
+                '--product-sku=sku2',
+            ],
+            $this->argumentBuilder->generateWithProductSku('product', 'store_1|store_2', 'sku1|sku2')
+        );
+    }
+
+    public function testGenerateWithProductSkusAll()
+    {
+        $this->loggerMock->expects($this->once())
+            ->method('info');
+
+        $this->assertEquals(
+            [
+                '--entity-type=product',
+                '--store-id=store_1',
+                '--store-id=store_2',
+            ],
+            $this->argumentBuilder->generateWithProductSku('product', 'store_1|store_2', '*')
```

Comment:
where is a log ?

Reply:
What do you mean?

Reply:
you call log function when sku is *. But you do not add this in test

Reply:
Actually I do check 
```
    public function testGenerateWithProductSkusAll()
    {
        $this->loggerMock->expects($this->once())
            ->method('info');
```

Reply:
oh, you are right :)

Reply:
But should not we test the message too?

Reply:
Added